src/sdkRuntimeModels.ts

```diff
@@ -217,14 +199,62 @@ export interface MParticleWebSDK {
         eventOptions?: SDKEventOptions
     ): void;
     logBaseEvent(event: BaseEvent, eventOptions?: SDKEventOptions): void;
+    logLink(selector: string, eventName: string, eventType: valueof<typeof EventType>, eventInfo: SDKEventAttrs): void;
+    logForm(selector: string, eventName: string, eventType: valueof<typeof EventType>, eventInfo: SDKEventAttrs): void;
+    logPageView(eventName: string, attrs?: SDKEventAttrs, customFlags?: SDKEventCustomFlags, eventOptions?: SDKEventOptions): void;
+    setOptOut(isOptingOut: boolean): void;
     eCommerce: SDKECommerceAPI;
-    logLevel: string;
-    ProductActionType: SDKProductActionType;
+
+    // QUESTION: Is this still used?
+    // logLevel: string;
+
+    ProductActionType: valueof<typeof ProductActionType>;
     generateHash(value: string): string;
+
+    // QUESTION: Should integration ID be a number or a string?
+    setIntegrationAttribute(integrationId: number, attrs: IntegrationAttribute): void;
+    getIntegrationAttributes(integrationId: number): IntegrationAttribute;
+}
+
+// https://go.mparticle.com/work/SQDSDKS-4805
+
+// https://go.mparticle.com/work/SQDSDKS-6949
+export interface IMParticleInstanceManager extends MParticleWebSDK {
+    // https://go.mparticle.com/work/SQDSDKS-5053
+    _BatchValidator: _BatchValidator;
+    _instances: Dictionary<IMParticleWebSDKInstance>;
+    _isTestEnv?: boolean;
+
+    EventType: valueof<typeof EventType>;
+    CommerceEventType: valueof<typeof CommerceEventType>;
+    PromotionType: valueof<typeof PromotionActionType>;
+    ProductActionType: valueof<typeof ProductActionType>;
+
     isIOS?: boolean;
-    sessionManager: Pick<ISessionManager, 'getSession'>; // https://go.mparticle.com/work/SQDSDKS-6949
+    identifyRequest: IdentityApiData;
```

Comment:
This is only used for tests and should be removed from interface. We may need to refactor tests. 😢